py/server/tests/test_udf_return.py

```diff
@@ -180,6 +180,15 @@ def fn3(col) -> List[Union[datetime.datetime, int]]:
             t = empty_table(1).update("X = i").update(f"Y= fn3(X + 1)")
             self.assertEqual(t.columns[1].data_type, dtypes.JObject)
 
+        with self.subTest("wrong typehint bool vs. np.bool_"):
+            def fn4(col: int) -> bool:
+                return np.bool_(col)
+
+            with self.assertRaises(DHError) as cm:
+                t = empty_table(2).update("X = i").update(f"Y= fn4(X + 1)")
+            self.assertIn("class org.jpy.PyObject cannot be cast to class java.lang.Boolean", str(cm.exception))
```

Comment:
I'm concerned about this case.  I'm guessing that >50% of users expect the numpy functions to return `bool` values.  With this error string, it doesn't give them enough information to know that the function is returning a `np.bool_` and not a `bool`.
1. Does the full error message clearly say that the return type is `np.bool_`.
2. Can the error message be amended to have the Python class of the object to make this easier to debug?
3. It sounds like we are missing a `np.bool_` -> `java.lang.Boolean` cast.  Does it make sense to add support for the cast?
4. Should the error message say that the returned value is not of the expected type?

Overall, I'm just trying to imagine what is going to happen in the wild, and I'm concerned about usability when things aren't on the ideal path.

Reply:
1. No, it is a runtime error from Java with no context of the UDF
2. We do have a ticket for that. #4098, not scheduled
3. filed #5397 
4. No, to do that we need to check the return value at runtime which is the exact thing we tried to avoid in the improvement PR.